src/config.rs

```diff
@@ -34,6 +36,30 @@ impl<T: HandshakeHandler> PeerNetConfiguration<T> {
             message_handlers: MessageHandlers::default(),
             fallback_function: None,
             handshake_handler,
+            optional_features: PeerNetFeatures::default(),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct PeerNetFeatures {
+    pub reject_same_ip_addr: bool,
+}
+
+impl Default for PeerNetFeatures {
+    fn default() -> PeerNetFeatures {
+        PeerNetFeatures {
+            reject_same_ip_addr: true,
+        }
+    }
+}
+
+impl PeerNetFeatures {
+    #[allow(clippy::needless_update)]
+    pub fn set_reject_same_ip_addr(self, val: bool) -> PeerNetFeatures {
```

Comment:
Can you provide some documentation on why you chose the pattern to create a new structure and return it instead of mutable reference ?

Reply:
Done